src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+  getArea: () => number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
 
-}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (
+      sideA >= sideB + sideC
+      || sideB >= sideA + sideC
+      || sideC >= sideA + sideB
+      || sideA <= 0
+      || sideB <= 0
+      || sideC <= 0
+    ) {
+      throw new Error('Invalid triangle');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const semiPerimeter = (this.sideA + this.sideB + this.sideC) * 0.5;
 
+    return +Math.sqrt(
+      semiPerimeter
+      * (semiPerimeter - this.sideA)
+      * (semiPerimeter - this.sideB)
+      * (semiPerimeter - this.sideC),
+    ).toFixed(2);
```

Comment:
in every case u use different way to round squre, u can do like this Math.floor(value * 100) / 100 and make the function outside to write less code